exampleCourse/questions/demo/annotated/engCircuit/server.py

```diff
@@ -1,40 +1,77 @@
 import random
+import io
 
+import matplotlib.pyplot as plt
+from schemdraw import Drawing
+import schemdraw.elements as elm
+
+def file(data):
+    if data["filename"] == "figure.png":
+        drawing = Drawing()
+        R1 = str(data["params"]["R1"]) + " $\\Omega$"
+        R2 = str(data["params"]["R2"]) + " $\\Omega$"
+        R3 = str(data["params"]["R3"]) + " $\\Omega$"
+        Vt = str(data["params"]["Vt"]) + "V"
+        if data["params"]["ask"] == "equivalent resistance $R_T$":
+            ## parallel resistors
+            drawing.push()
+            drawing += elm.Line()
+            drawing.push()
+            drawing += elm.Resistor().down().label(R1)
+            drawing.pop()
+            drawing += elm.Line().right()
+            drawing.push()
+            drawing += elm.Resistor().down().label(R2)
+            drawing.pop()
+            drawing += elm.Line().right()
+            drawing += elm.Resistor().down().label(R3)
+            drawing += elm.Line().left()
+            drawing += elm.Line().left()
+            drawing += elm.Line().left()
+            drawing.pop()
+            drawing += elm.BatteryCell().down().label(Vt,'bottom')
+        else:
+            ## Series resistors
+            drawing.push()
+            drawing += elm.Line().right()
+            drawing += elm.Line().right()
+            drawing += elm.Line().right()
+            drawing += elm.Line().down()
+            drawing += elm.Resistor().left().label(R3)
+            drawing += elm.Resistor().left().label(R2)
+            drawing += elm.Resistor().left().label(R1)
+            drawing.pop()
+            drawing += elm.BatteryCell().down().label(Vt,'bottom')
+        drawing.draw(show=False)
+        buf = io.BytesIO()                       
+        plt.savefig(buf, format="png")  
+        return buf
```

Comment:
I don't think you need matplotlib here. As per documentation you can return a byte array, which is what you get with `get_image_data` directly.
```suggestion
        return drawing.get_image_data()
```

Reply:
Yes, `drawing.get_image_data()` does return a byte array but the browser cannot render it directly.
I can see two options from here:
1) use the matplotlib method which would align with the [documentation](https://prairielearn.readthedocs.io/en/latest/question/#generating-dynamic-files)
2) use `get_image_data` and convert it into render able image using another library

Which way do you suggest? or perhaps I missed something.

Reply:
I just tested locally, and there is a typo in my suggestion, sorry. It should be `get_imagedata` instead of `get_image_data`. This change resolved the issue you mentioned. In any case, to avoid problems, I'd suggest you change the image name to `figure.svg` instead of `figure.png`, both in the test in `file` and in the `pl-figure` option.

Reply:
Done here https://github.com/PrairieLearn/PrairieLearn/pull/8179/commits/cee16421d3b9d18c227a088f4ec593afb689c018